attestation/secretscan/secretscan.go

```diff
@@ -0,0 +1,449 @@
+		if len(gf.Secret) > 8 {
+			truncatedSecret = gf.Secret[:4] + "..." + gf.Secret[len(gf.Secret)-4:]
+		} else if len(gf.Secret) > 0 {
+			truncatedSecret = "..."
+		}
+
+		finding := Finding{
+			RuleID:       gf.RuleID,
+			Description:  gf.Description,
+			Severity:     getDefaultSeverity(gf.RuleID),
+			File:         filePath,
+			Line:         gf.StartLine,
+			Match:        gf.Match,
+			Secret:       truncatedSecret,
+			Entropy:      gf.Entropy,
+			TruncatedKey: len(gf.Secret) > 0,
+		}
+		findings = append(findings, finding)
+	}
+
+	return findings, nil
+}
+
+// getDefaultSeverity returns a default severity based on the rule ID
+func getDefaultSeverity(ruleID string) string {
+	// These are very common high-severity rule types
+	highSeverityPrefixes := []string{
+		"aws", "gcp", "azure", "api", "token", "key", "credential", "password",
+		"secret", "private", "auth", "ssh", "cert", "jwt",
+	}
+
+	for _, prefix := range highSeverityPrefixes {
+		if strings.Contains(strings.ToLower(ruleID), prefix) {
+			return "HIGH"
+		}
+	}
+
+	return "MEDIUM"
+}
+
+// isBinaryFile returns true if the MIME type represents a binary file
+func isBinaryFile(mimeType string) bool {
+	// Common binary mime types
+	binaryPrefixes := []string{
+		"application/octet-stream",
+		"application/x-executable",
+		"application/x-mach-binary",
+		"application/x-sharedlib",
+		"application/x-object",
+	}
+
+	for _, prefix := range binaryPrefixes {
+		if strings.HasPrefix(mimeType, prefix) {
+			return true
+		}
+	}
+
+	// Executable files like .exe, .dll, .so
+	executableSuffixes := []string{
+		"/x-executable",
+		"/x-sharedlib",
+		"/x-mach-binary",
+	}
+
+	for _, suffix := range executableSuffixes {
+		if strings.HasSuffix(mimeType, suffix) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// extractStringsFromBinary extracts printable strings from a binary file
+func extractStringsFromBinary(binaryPath, outputPath string) error {
+	// Read the binary file
+	content, err := os.ReadFile(binaryPath)
+	if err != nil {
+		return err
+	}
+
+	// Extract ASCII strings (basic implementation)
+	var stringsList []string
+	var currentString []byte
+	for _, b := range content {
+		// ASCII printable characters
+		if b >= 32 && b <= 126 {
+			currentString = append(currentString, b)
+		} else if len(currentString) >= 8 { // Only keep strings of reasonable length
+			stringsList = append(stringsList, string(currentString))
+			currentString = nil
+		} else {
+			currentString = nil
+		}
+	}
+	if len(currentString) >= 8 {
+		stringsList = append(stringsList, string(currentString))
+	}
+
+	// Write strings to output file
+	return os.WriteFile(outputPath, []byte(strings.Join(stringsList, "\n")), 0644)
+}
+
+func (a *Attestor) MarshalJSON() ([]byte, error) {
+	type alias Attestor
+	return json.Marshal(&struct {
+		*alias
+		Findings []Finding `json:"findings"`
+	}{
+		alias:    (*alias)(a),
+		Findings: a.Findings,
+	})
+}
+
+func (a *Attestor) UnmarshalJSON(data []byte) error {
```

Comment:
Similar to above, this doesn't seem to add anything that the default implementations of Marshal/Unmarshal would give us